src/librustdoc/html/static/js/search.js

```diff
@@ -4919,11 +4919,12 @@ ${item.displayPath}<span class="${type}">${name}</span>\
         });
     } else if (query.error === null) {
         const dlroChannel = `https://doc.rust-lang.org/${getVar("channel")}`;
+        const docsrsURL = `https://docs.rs/releases/search?query=${encodeURIComponent(query.userQuery)}`;
+        const duckduckgoURL = `https://duckduckgo.com/?q=${encodeURIComponent("rust " + query.userQuery)}`;
         output.className = "search-failed" + extraClass;
-        output.innerHTML = "No results :(<br/>" +
-            "Try on <a href=\"https://duckduckgo.com/?q=" +
-            encodeURIComponent("rust " + query.userQuery) +
-            "\">DuckDuckGo</a>?<br/><br/>" +
+        output.innerHTML = `No results in <code>${getVar("current-crate")}</code> :(<br/>` +
+            `Search for <a href="${docsrsURL}">other crates</a>?<br/>` +
+            `Or try on <a href="${duckduckgoURL}">DuckDuckGo</a>?<br/><br/>` +
```

Comment:
we can apply some simple heuristics to hide this when it would be unhelpful.  for example, if `query.hasReturnArrow` is true, we're doing type-based search, like `u8 -> u16`, which probably won't give much help.